java7/Dockerfile

```diff
@@ -0,0 +1,22 @@
+FROM ubuntu:12.04
+ENV DEBIAN_FRONTEND noninteractive
+
+# Accept oracle license in debconf
+ADD java7.debconf /tmp/java7.debconf
+# Add webupdate8 apt repo
+ADD java7.list /etc/apt/sources.list.d/java7.list
+# Add webupdate8 signing key
+RUN apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 7B2C3B0889BF5709A105D03AC2518248EEA14886
+RUN apt-get update -qq
+
+RUN cat /tmp/java7.debconf |debconf-set-selections
+RUN apt-get install -qy oracle-java7-installer
+RUN rm -f /var/cache/oracle-jdk7-installer/jdk*tar.gz
```

Comment:
It won't free space because it exists in the previous docker layer.

The usual approach is to chain apt related commands in a single RUN statement like you did for maven.